### PR TITLE
Refactor `NwbImagingExtractor` to support extracting data from `OnePhotonSeries`

### PR DESCRIPTION
**Pull request by weiglszonja**
Allow extracting data from `OnePhotonSeries` when using `NwbImagingExtractor` for testing.

### CODE REVIEW

File: src/roiextractors/extractors/nwbextractors/nwbextractors.py [hunk 1]
```diff
@@ -73,33 +77,37 @@ def __init__(self, file_path: PathType, optical_series_name: Optional[str] = "Tw
...
+        self.photon_series = self.nwbfile.acquisition[self._optical_series_name]
+        photon_series = dict(
+            OnePhotonSeries=OnePhotonSeries,
+            TwoPhotonSeries=TwoPhotonSeries,
+        )[self._optical_series_name]
         assert isinstance(
-            self.two_photon_series, TwoPhotonSeries
-        ), "The optical series must be of type pynwb.TwoPhotonSeries"
+            self.photon_series, photon_series
+        ), f"The optical series must be of type {photon_series.__name__}"
```
Comment (weiglszonja) [inline src/roiextractors/extractors/nwbextractors/nwbextractors.py:87]: change to is at least OnePhoton or TwoPhoton, the optical series name can be anything not necessarily a valid photon series type so need to change that.

File: src/roiextractors/extractors/nwbextractors/nwbextractors.py [hunk 2]
```diff
@@ -49,14 +49,18 @@ class NwbImagingExtractor(ImagingExtractor):
     mode = "file"
     installation_mesg = "To use the Nwb Extractor run:\n\n pip install pynwb\n\n"  # error message when not installed
 
-    def __init__(self, file_path: PathType, optical_series_name: Optional[str] = "TwoPhotonSeries"):
+    def __init__(
+        self,
+        file_path: PathType,
+        optical_series_name: Literal["TwoPhotonSeries", "OnePhotonSeries"] = "TwoPhotonSeries",
```
Comment (weiglszonja) [inline src/roiextractors/extractors/nwbextractors/nwbextractors.py:55]: This is just the name, no need to change
```suggestion
        optical_series_name: Optional[str] = "TwoPhotonSeries"
```

File: src/roiextractors/extractors/nwbextractors/nwbextractors.py [hunk 3]
```diff
@@ -49,14 +49,18 @@ class NwbImagingExtractor(ImagingExtractor):
...
+    ):
         """
         Parameters
         ----------
         file_path: str
             The location of the folder containing dataset.nwb file
-        optical_series_name: str (optional)
-            optical series to extract data from
+        optical_series_name: {'TwoPhotonSeries', 'OnePhotonSeries'}, optional
+            The optical series to extract data from.
```
Comment (weiglszonja) [inline src/roiextractors/extractors/nwbextractors/nwbextractors.py:63]:
```suggestion
        optical_series_name: str (optional)
            The name of the optical series to extract data from.
```

File: src/roiextractors/extractors/nwbextractors/nwbextractors.py [hunk 4]
```diff
@@ -56,7 +56,7 @@ def __init__(self, file_path: PathType, optical_series_name: Optional[str] = "Tw
         file_path: str
             The location of the folder containing dataset.nwb file
         optical_series_name: str (optional)
```
Comment (CodyCBakerPhD) [inline src/roiextractors/extractors/nwbextractors/nwbextractors.py:58]:
```suggestion
        optical_series_name: string, optional
```

This is closer to numpy docstring style right? Not that we've updated that everywhere on the repo though...

Comment (CodyCBakerPhD) [reply src/roiextractors/extractors/nwbextractors/nwbextractors.py:58]: Since the rest of this looks good, just gonna push it through

**Review by CodyCBakerPhD — APPROVED** [review]

**Pull request merged**